parsers/AR.py

```diff
@@ -23,957 +22,111 @@
-            # not a string....
-            continue
-
-    hydro_generation = generation_finder(mapped_data, 'hydro')
-    solar_generation = generation_finder(mapped_data, 'solar')
-    wind_generation = generation_finder(mapped_data, 'wind')
-    unknown_generation = generation_finder(mapped_data, 'unknown')
-    hydro_storage_generation = generation_finder(mapped_data, 'hydro_storage')
-
-    return {'hydro': hydro_generation,
-            'wind': wind_generation,
-            'solar': solar_generation,
-            'unknown': unknown_generation,
-            'hydro_storage': hydro_storage_generation}
-
-
-def fetch_production(zone_key='AR', session=None, target_datetime=None, logger=logging.getLogger(__name__)) -> dict:
-    """Requests the last known production mix (in MW) of a given country."""
-    if target_datetime is not None:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
-
-    gdt = get_datetime(session=None)
-    thermal = get_thermal(session, logger)
-    hydro = get_hydro_and_renewables(session, logger)
-
-    # discharging is given positive value in data, opposite to EM
-    hydro_storage = hydro.pop('hydro_storage')
-    if hydro_storage == 0.0:
-        em_hydro_storage = hydro_storage
-    else:
-        em_hydro_storage = hydro_storage*-1
-
-    unknown = thermal.pop('unknown') + hydro.pop('unknown')
-    production = {**hydro, **thermal}
-    production['unknown'] = unknown
-
-    production_mix = {
-        'zoneKey': zone_key,
-        'datetime': gdt['datetime'],
-        'production': production,
-        'storage': {
-            'hydro': em_hydro_storage,
-        },
-        'source': 'portalweb.cammesa.com'
+    
+    return renewables_production
+
+def non_renewables_production_mix(zone_key: str, session) -> Dict[str, dict]:
+    """Retrieves production mix for non renewables using CAMMESA's API"""
+
+    params = {'id_region': 1002}
+    api_cammesa_response = session.get(CAMMESA_DEMANDA_ENDPOINT, params=params)
+    assert api_cammesa_response.status_code == 200, 'Exception when fetching production for ' \
+                                   '{}: error when calling url={} with payload={}'.format(
+                                       zone_key, CAMMESA_DEMANDA_ENDPOINT, params)
+
+    production_list = api_cammesa_response.json()
+    sorted_production_list = sorted(production_list, key=lambda d: d['fecha'])
+    
+
+    non_renewables_production: Dict[str, dict] = {
+        production_info['fecha']:
+        {
+            'hydro': production_info['hidraulico'],
+            'nuclear': production_info['nuclear'],
+            # As of 2022 thermal energy is mostly natural gas but 
+            # the data is not split. We put it into unknown for now.
+            # More info: see page 21 in https://microfe.cammesa.com/static-content/CammesaWeb/download-manager-files/Sintesis%20Mensual/Informe%20Mensual_2021-12.pdf
+            'unknown': production_info['termico']
```

Comment:
Should we just put all of the thermal into 'gas' or leave it to 'unknown'? It's a pity to put that into unknown because it accounts for most of the energy production.

Reply:
For now I think it's better to keep it as unknown, even though it would be great to attribute it more appropriately. Would be great to reach out to the source and ask if there's any way they could consider splitting it up in their API :)
(But that's a next step we can do afterwards)